pkg/database/user_stats.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package database
+
+import (
+	"time"
+
+	"github.com/jinzhu/gorm"
+)
+
+// UserStats represents statistics related to a user in the database.
+type UserStats struct {
+	gorm.Model
+	Date        time.Time `gorm:"unique_index:idx_date_user_realm"`
+	UserID      int64     `gorm:"unique_index:idx_date_user_realm"`
+	User        User
+	RealmID     int64 `gorm:"unique_index:idx_date_user_realm"`
+	Realm       Realm
```

Comment:
same - I think we just need the RealmID for tracking but don't need to traverse the models from the stats level.